core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -1839,3 +1890,354 @@ oppia.factory('explorationWarningsService', [
     };
   }
 ]);
+
+oppia.factory('humanReadableLostChangesService', function() {
+  var TYPE_ADD_STATE = 'add_state';
+  var TYPE_DELETE_STATE = 'delete_state';
+  var TYPE_RENAME_STATE = 'rename_state';
+  var TYPE_EDIT_STATE_PROPERTY = 'edit_state_property';
+
+  var TYPE_CONTENT = 'content';
+
+  var TYPE_ADD_INTERACTION = 'add_interaction';
+  var TYPE_DELETE_INTERACTION = 'delete_interaction';
+  var TYPE_END_EXPLORATION = 'end_exploration';
+
+  var TYPE_ADD_WIDGET_CUSTOMIZATIONS = 'add_widget_customizations';
+  var TYPE_DELETE_WIDGET_CUSTOMIZATIONS = 'delete_widget_customizations';
+  var TYPE_EDIT_WIDGET_CUSTOMIZATIONS = 'edit_widget_customizations';
+
+  var TYPE_ADD_ANSWER_GROUP = 'add_answer_group';
+  var TYPE_DELETE_ANSWER_GROUP = 'delete_answer_group';
+  var TYPE_EDIT_ANSWER_GROUP = 'edit_answer_group';
+
+  var TYPE_RULE = 'rule';
+
+  var TYPE_ADD_DEFAULT_OUTCOME = 'add_default_outcome';
+  var TYPE_DELETE_DEFAULT_OUTCOME = 'delete_default_outcome';
+  var TYPE_EDIT_DEFAULT_OUTCOME = 'edit_default_outcome';
+
+  var _createChangeObject = function(label, type, value, stateName) {
```

Comment:
You don't need underscores in front of all these names as they are inherently local.